docs/proposals/config_data.md

```diff
@@ -0,0 +1,319 @@
+          downwardAPI:
+            items:
+              - path: "config.yaml"
+                configDataRef:
+                  from: PodConfigurationVol
+                  fieldPath: component_config
+              - path: "metadata.json"
+                configDataRef:
+                  from: PodConfigurationVol
+                  fieldPath: component_metadata
+```
+
+Now, we create the replication controller:
+```bash
+kubectl create -f example-replication-controller.yaml
+```
+
+New pods created by the ExampleReplicationController will use the ``PodConfigurationVol`` objects when starting/replicating pods.
+
+``PodConfigurationVol`` object will be watched for any modifications and the mounted volumes will be updated if there are any modifications.
+
+#### Consuming ConfigData as environment variables
+
+```json
+{
+    "apiVersion": "v1",
+    "name": "PodConfigurationEnv",
+    "kind": "ConfigData",
+    "data": {
+        "api_server": "10.240.13.14:8998",
+        "api_version": "v1",
+        "db_host": "10.240.55.3",
+        "db_port": "3306"
+    }
+}
+```
+``PodConfigurationEnv`` is intended to be consumed as environment variable key/value pairs.
+
+We create our ``ConfigData`` object.
+```bash
+kubectl create -f pod-configuration-env.json
+```
+
+ExampleReplicationController starts 2 replicas that runs the `busybox` container.
+In this use case, the ``ConfigData`` object will be consumed as 4 environment variables.
+```yaml
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: ExampleReplicationController
+spec:
+  replicas: 2
+  selector:
+    app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: busybox
+        image: gcr.io/google_containers/busybox
+        ports:
+        - containerPort: 80
+        env:
+        - name: API_SERVER
+          valueFrom:
+            configDataRef:
+              from: PodConfigurationEnv
+              key: api_servers
+        - name: API_VERSION
+          valueFrom:
+            configDataRef:
+              from: PodConfigurationEnv
+              key: api_version
+        - name: DB_HOST
+          valueFrom:
+            configDataRef:
+              from: PodConfigurationEnv
+              key: postgresql_host
+        - name: DB_PORT
+          valueFrom:
+            configDataRef:
+              from: PodConfigurationEnv
+              key: postgresql_port
+```
+
+Now, we create the replication controller:
+```bash
+kubectl create -f example-replication-controller.yaml
+```
+
+New pods created by the ``ExampleReplicationController`` will use the ``PodConfigurationEnv`` objects when starting/replicating pods.
+
+``PodConfigurationEnv`` object will be watched for any modifications and the currently consumed environment variables will be updated.
+
+### Future Improvements
+
+* Additional specification parameters can be added to the API for describing the expected behavior after
```

Comment:
@pmorie True. But, that part is about refreshing the consumed objects and not about restarting consumers. Since, we decided to postpone the consumer restarts, I included it in the future improvements section.